src/content_script.js

```diff
@@ -173,120 +167,203 @@ function createBadgeFragment (socialAction) {
+    // Flip the iframe to show on the left side when icon is too close to the edge
+    if (iframePaddingAllowance > calculateOffsetDiff) {
+      iframeBox.style.marginLeft = `${xLeft}px`;
+      iframeChevron.classList.add("fbc-chevron-arrow-right");
+      iframeChevron.style.marginLeft = `${xPosChevron - fencePos.offsetWidth - iframeChevron.offsetWidth - offsetX}px`;
     }
 
-    htmlBadgeFragmentPromptDiv.appendChild(htmlBadgeFragmentPromptButtonDiv);
+    else {
+      iframeChevron.classList.remove("fbc-chevron-arrow-right");
+    }
+
+    iframeChevron.classList.remove("fbc-chevron-arrow-top");
   }
 
-  // Create Empty Wrapper Div
-  const htmlBadgeWrapperDiv = document.createElement("div");
-  htmlBadgeWrapperDiv.appendChild(htmlBadgeFragment);
+  // Mobile Orientation
+  else {
+    for (const panels of iframeElement) {
+      panels.width = window.innerWidth;
+      if (window.innerWidth > 480) {
+        panels.width = 350;
+      }
+    }
 
-  return htmlBadgeWrapperDiv;
+    iframeChevron.classList.add("fbc-chevron-arrow-top");
+    iframeBox.style.marginTop = `${yPosMobile}px`;
+
+    const xPosChevronMobile = xPosMobile;
+    const yPosChevronMobile = yPosMobile - iframeChevron.offsetWidth;
+
+    iframeChevron.style.marginLeft = `${xPosChevronMobile}px`;
+    iframeChevron.style.marginTop = `${yPosChevronMobile}px`;
+
+  }
 }
 
-function shouldBadgeBeSmall(ratioCheck, itemHeight) {
-  if (ratioCheck < 1.1) {
-    return true;
-  } else if (itemHeight < 39) {
-    return true;
+function openLoginPrompt(socialAction, fencePos, target, FBC_IFRAME_HEIGHT) {
+
+  const iframeSrcVal = buildInpageIframe(socialAction, target, FBC_IFRAME_HEIGHT).src;
+
+  const hasFbcWrapper = document.querySelector(".fbc-wrapper");
+  if (!hasFbcWrapper) {
+    document.body.appendChild(injectIframeOntoPage(socialAction, target, FBC_IFRAME_HEIGHT));
+    positionIframe(fencePos);
+    ["resize", "scroll"].forEach(function (evt) {
+      if (document.querySelector(".fbc-wrapper")) {
+        window.addEventListener(evt, () => {
+          positionIframe(fencePos);
+        });
+      }
+    });
+
+    postMessageListeners(iframeSrcVal, target);
+    
+  } else {
+    hasFbcWrapper.remove();
   }
-  return false;
 }
 
-function openLoginPrompt(parent, el, htmlBadgeDiv) {
-  parent.classList.toggle("active");
-  positionPrompt( htmlBadgeDiv );
-  el.classList.toggle("js-fbc-prompt-active");
-  document.body.classList.toggle("js-fbc-prompt-active");
-  htmlBadgeDiv.querySelector(".fbc-badge-prompt-btn-cancel").focus();
+function postMessageListeners(iframeSrcVal, target){
+
+  window.addEventListener("message", (e) => {
+    if (
+      e.data === "allowTriggered" 
+      && iframeSrcVal.includes(e.origin)
+    ){
+      target.click();
+    }
+  });
+
+  window.addEventListener("message", (e) => {
+    if (
+      e.data === "closeTheInjectedIframe" 
+      && iframeSrcVal.includes(e.origin)
+    ) {
+      closeIframe();
+    }
+  });
 }
```

Comment:
@groovecoder I was wondering what you thought about keeping the `postMessage` method if I specified the origin is the `iframe` source value. I did more read up about it on [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage) and I saw this:

> 
If you do expect to receive messages from other sites, always verify the sender's identity using the origin and possibly source properties.

Reply:
Hmm ... that sounds good to me - if it works it certainly keeps the code simpler than running thru the background script.

Can you try it that way and make sure it still stops the original POC "attack"?